staramr/results/AMRDetectionSummary.py

```diff
@@ -42,37 +48,105 @@ def _compile_plasmids(self, ds):
+
+            if not plasmid_frame.empty:
+                set_used = negative_plasmid_names_set
+
+        negative_plasmid_entries = pd.DataFrame([[x, 'None'] for x in set_used],
+                                                columns=('Isolate ID', 'Gene')).set_index('Isolate ID')
+        negative_plasmid_entries['Data Type']='Plasmid'
+
+        if negative_entries is None:
+            negative_entries = negative_plasmid_entries
+        else:
+            negative_entries = negative_entries.append(negative_plasmid_entries, sort=True)
+
+        return resistance_frame.append(negative_entries, sort=True)
+
+    def create_summary(self, include_negatives: bool=False) -> DataFrame:
         """
-        Constructs a summary pd.DataFrame for all ResFinder/PointFinder results.
-        :param include_negatives: If True, include files with no ResFinder/PointFinder results.
+        Constructs a summary pd.DataFrame for all ResFinder/PointFinder/PlasmidFinder results.
+        :param include_negatives: If True, include files with no ResFinder/PointFinder/PlasmidFinder results.
         :return: A pd.DataFrame summarizing the results.
         """
-        df = self._resfinder_dataframe
-        ds = self._plasmidfinder_dataframe
+        resistance_frame = self._resfinder_dataframe
+        plasmid_frame = self._plasmidfinder_dataframe
 
         if self._has_pointfinder:
-            df = df.append(self._pointfinder_dataframe, sort=True)
+            resistance_frame = resistance_frame.append(self._pointfinder_dataframe, sort=True)
 
-        df = self._compile_results(df)
+        resistance_frame = self._compile_results(resistance_frame)
 
         if include_negatives:
-            df = self._include_negatives(df)
+            resistance_frame = self._include_negatives(resistance_frame)
+
+        resistance_frame.rename(columns={'Gene': 'Genotype'}, inplace=True)
 
-        df.rename(columns={'Gene': 'Genotype'}, inplace=True)
+        if plasmid_frame is not None:
+            plasmid_frame = self._compile_plasmids(plasmid_frame)
 
-        if ds is not None:
-            ds = self._compile_plasmids(ds)
+            resistance_frame = resistance_frame.merge(plasmid_frame, on='Isolate ID', how='left').fillna(value={'Plasmid Genes': 'None'})
 
-            df = df.merge(ds, on='Isolate ID', how='left').fillna(value={'Plasmid Genes': 'None'})
+        return resistance_frame.sort_index()
 
-        return df.sort_index()
+    def create_detailed_summary(self, include_negatives: bool=True) -> DataFrame:
+        resistance_frame = self._resfinder_dataframe
+        resistance_frame['Data Type']='Resistance'
+        plasmid_frame = self._plasmidfinder_dataframe
+
+        column_names = ['Gene', 'Predicted Phenotype','%Identity', '%Overlap', 'HSP Length/Total Length','Contig', 'Start', 'End', 'Accession', 'Data Type']
+
+        if self._has_pointfinder:
+            point_frame = self._pointfinder_dataframe
```

Comment:
The same as my comment above about making a copy of `self._resfinder_dataframe`.

Reply:
Resolved in b2bd0a3